packages/migrate/src/__tests__/DbPull/schema-folder.test.ts

```diff
@@ -0,0 +1,269 @@
+    "config.prisma:
+
+    generator client {
+      provider        = "prisma-client-js"
+      output          = "@prisma/client"
+      previewFeatures = ["prismaSchemaFolder"]
+    }
+
+    datasource my_db {
+      provider = "sqlite"
+      url      = "file:dev.db"
+    }
+
+
+    introspected.prisma:
+
+    model Blog {
+      id          Int  @id @default(autoincrement())
+      viewCount20 Int
+      ownerId     Int
+      User        User @relation(fields: [ownerId], references: [id])
+    }
+
+
+    User.prisma:
+
+    model User {
+      id   Int    @id @default(autoincrement())
+      Blog Blog[]
+    }
+    "
+  `)
+})
+
+test('reintrospection - new model - existing introspected.prisma', async () => {
+  ctx.fixture('introspection-folder-new-model-with-introspected')
+  const introspect = new DbPull()
+  const result = introspect.parse([])
+  await expect(result).resolves.toMatchInlineSnapshot(`""`)
+
+  expect(ctx.printDir('prisma/schema', ['.prisma'])).toMatchInlineSnapshot(`
+    "config.prisma:
+
+    generator client {
+      provider        = "prisma-client-js"
+      output          = "@prisma/client"
+      previewFeatures = ["prismaSchemaFolder"]
+    }
+
+    datasource my_db {
+      provider = "sqlite"
+      url      = "file:dev.db"
+    }
+
+
+    introspected.prisma:
+
+    model User {
+      id   Int    @id @default(autoincrement())
+      Blog Blog[]
+    }
+
+    model Blog {
+      id          Int  @id @default(autoincrement())
+      viewCount20 Int
+      ownerId     Int
+      User        User @relation(fields: [ownerId], references: [id])
+    }
+    "
+  `)
+})
+
+test('reintrospection - new field', async () => {
+  ctx.fixture('introspection-folder-new-field')
+  const introspect = new DbPull()
+  const result = introspect.parse([])
+  await expect(result).resolves.toMatchInlineSnapshot(`""`)
+
+  expect(ctx.printDir('prisma/schema', ['.prisma'])).toMatchInlineSnapshot(`
+    "Blog.prisma:
+
+    model Blog {
+      id          Int  @id @default(autoincrement())
+      viewCount20 Int
+      ownerId     Int
+      owner       User @relation(fields: [ownerId], references: [id])
+    }
+
+
+    config.prisma:
+
+    generator client {
+      provider        = "prisma-client-js"
+      output          = "@prisma/client"
+      previewFeatures = ["prismaSchemaFolder"]
+    }
+
+    datasource my_db {
+      provider = "sqlite"
+      url      = "file:dev.db"
+    }
+
+
+    User.prisma:
+
+    model User {
+      id    Int    @id @default(autoincrement())
+      blogs Blog[]
+    }
+    "
+  `)
+})
+
+test('reintrospection - remove model', async () => {
+  ctx.fixture('introspection-folder-remove-model')
+  const introspect = new DbPull()
+  const result = introspect.parse([])
+  await expect(result).resolves.toMatchInlineSnapshot(`""`)
+
+  expect(ctx.printDir('prisma/schema', ['.prisma'])).toMatchInlineSnapshot(`
+    "Blog.prisma:
+
+
+
+
+    config.prisma:
```

Comment:
The comments `//` are removed from the snapshot?

Reply:
This is not `--print` output, this use snapshoting the contents of the directory. See `ctx.printDir`